### ISSUE TITLE
rsun_obs needs to be updated in differential_rotation for new observer

### ISSUE THREAD

**Issue by hayesla** [sunpy/sunpy#3202]
The output map from the `sunpy.physics.differential_rotation.differential_rotation` has the same `rsun_obs` when the observer location is changed. This is needed for `draw_limb` 

Example :
```
>>> import sunpy.map
>>> from sunpy.physics.differential_rotation import differential_rotate
>>> from sunpy.coordinates.ephemeris import get_body_heliographic_stonyhurst
>>> aia_map = sunpy.map.Map('/Users/laurahayes/notebooks_tests/aia_lev1_131a_2014_10_20t00_00_08_62z_image_lev1.fits')
>>> new_map = differential_rotate(aia_map, observer = get_body_heliographic_stonyhurst(body = 'mars', time=aia_map.date))

>>> new_map.plot()
>>> new_map.draw_limb(color  = 'k')
```
![test](https://user-images.githubusercontent.com/4620298/59208423-af95bc00-8b6e-11e9-995b-66ca9f1dae59.png)

I think it just needs to be fit into new observer meta `out_meta` in `sunpy.physics.differential_rotation`

**Comment by Cadair** [comment]
I think this is fixed by #3225

**Comment by Cadair** [comment]
Yes, I confirmed this is now fixed.
